include/sharg/detail/format_parse.hpp

```diff
@@ -172,32 +172,47 @@ class format_parse : public format_base
      * The `id` is found by comparing every argument in argv to `id` prepended with two dashes (`--`)
      * or a prefix of such followed by the equal sign `=`.
      */
-    template <typename iterator_type, typename id_type>
-    static iterator_type find_option_id(iterator_type begin_it, iterator_type end_it, id_type const & id)
+    template <typename iterator_type>
+    static iterator_type find_option_id(iterator_type begin_it, iterator_type end_it, id_pair const & ids)
     {
-        if (is_empty_id(id))
+        bool short_id_empty{ids.short_id.empty()};
+        bool long_id_empty{ids.long_id.empty()};
+
+        if (short_id_empty && long_id_empty)
```

Comment:
```suggestion
        assert(short_id_empty || long_id_empty); // one id has to be non-empty
```

Is this possible? I would bet that we catch this design error somewhere as it makes no sense. This kind of option would never be parsed.

Reply:
I'll have to check :D

Reply:
Nope, it's used here in a way that expects to find nothing:

https://github.com/seqan/sharg-parser/blob/39f65a4890f8c5108af2b5c7974893ff6ed87e50/include/sharg/detail/format_parse.hpp#L677-L681